### PR TITLE
🏃 Add unit tests for health check function

### PR DESCRIPTION
**Pull request by chuckha**
Signed-off-by: Chuck Ha <chuckh@vmware.com>

This codifies (in tests) the expectations we make around meta health checks.

Here are a list of checks included in this PR:

- When a machine's node was not checked for health. An absence of health from a node is bad.
- The health check returns an error not related to a node's health. For example, the health check cannot connect to the api server to get a list of nodes.
- More nodes than machines were checked. This means there are control plane nodes not managed by cluster api and that is considered unhealthy.
- A machine has a nil node reference. This means the control plane hasn't actually been created or is in the process of being deleted. The control plane cannot be healthy.

cc @dlipovetsky If i'm missing anything you can think of, please let me know! 

Somewhat related to #2454

### CODE REVIEW

File: controlplane/kubeadm/internal/cluster_test.go [hunk 1]
```diff
@@ -242,3 +243,161 @@ func (f *fakeClient) List(_ context.Context, list runtime.Object, _ ...client.Li
...
+					controlPlaneMachine("two"),
+					controlPlaneMachine("three"),
+				},
+			},
+			check: func(ctx context.Context) (healthCheckResult, error) {
+				return healthCheckResult{
+					"one":   nil,
+					"two":   errors.New(""),
+					"three": errors.New(""),
+				}, errors.New("")
```
Comment (sethp-nr) [inline controlplane/kubeadm/internal/cluster_test.go]: The problem I ran into specifically was that in this case, the "two" and "three" errors were being swallowed by the healthcheck function. 

wdyt of `errors.New("two")`, `errors.New("three")`, `errors.New("meta")`, and checking that each string (`two`, `three`, `meta`) shows up in the error.Error() string?

Comment (sethp-nr) [reply controlplane/kubeadm/internal/cluster_test.go]: Also, possibly worth covering is the case where "two" and "three" have errors, but the "overall" error is nil? I'm not quite sure what the expectation would be in that case.

Comment (chuckha) [reply controlplane/kubeadm/internal/cluster_test.go]: Ah ha. Perfect. Let me play around a bit. Thank you for reviewing!

**Comment by chuckha** [comment]
@sethp-nr alright, case is covered!

**Comment by sethp-nr** [comment]
Awesome,  thanks Chuck!

/lgtm (though I don't think I have the bit)

**Comment by detiber** [comment]
/lgtm
@sethp-nr we should probably get you added as a reviewer, then :grin:

**Comment by chuckha** [comment]
@sethp-nr you can /lgtm since you're in the kubernetes-sigs org (i'm like 99% sure) 

but you have to put the /lgtm alone on it's own line

**Comment by sethp-nr** [comment]
@chuckha Oh, I'll have to try it properly next time.

@detiber I certainly wouldn't say no :)

**Pull request merged**